internal/tasks/repotester/repotester.go

```diff
@@ -59,6 +73,100 @@ func (r *RepoTester) TestRepositories() {
 	}
 }
 
+func configureRepoHTTPSClient(httpConfig api.RepositoryHttpConfig) error {
+	tlsConfig := tls.Config{} //nolint:gosec
+	if httpConfig.SkipServerVerification != nil {
+		tlsConfig.InsecureSkipVerify = *httpConfig.SkipServerVerification //nolint:gosec
+	}
+
+	if httpConfig.TlsClientCertData != nil && httpConfig.TlsClientCertKey != nil {
+		cert, err := b64.StdEncoding.DecodeString(*httpConfig.TlsClientCertData)
+		if err != nil {
+			return err
+		}
+
+		key, err := b64.StdEncoding.DecodeString(*httpConfig.TlsClientCertKey)
+		if err != nil {
+			return err
+		}
+
+		tlsPair, err := tls.X509KeyPair(cert, key)
+		if err != nil {
+			return err
+		}
+
+		tlsConfig.Certificates = []tls.Certificate{tlsPair}
+	}
+
+	if httpConfig.RootCA != nil {
+		ca, err := b64.StdEncoding.DecodeString(*httpConfig.RootCA)
+		if err != nil {
+			return err
+		}
+
+		rootCAs, _ := x509.SystemCertPool()
+		if rootCAs == nil {
+			rootCAs = x509.NewCertPool()
+		}
+		rootCAs.AppendCertsFromPEM(ca)
+		tlsConfig.RootCAs = rootCAs
+	}
+
+	gitclient.InstallProtocol("https", githttp.NewClient(
+		&http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tlsConfig,
+			},
+		},
+	))
+	return nil
+}
+
+func GetAuth(repository *model.Repository) (transport.AuthMethod, error) {
```

Comment:
since this is public can we add some basic docs?

Reply:
Docs added. Let me know if it's not enough.